### PR TITLE
[Efficient Metadata Operations] Add metric to track slipped puts for metadata chunks.

### PR DESCRIPTION
**Pull request by ankagrawal**

### CODE REVIEW

File: ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouterMetrics.java [hunk 1]
```diff
@@ -184,7 +184,9 @@ public class NonBlockingRouterMetrics {
   public final Meter operationErrorRate;
   public final Meter encryptedOperationErrorRate;
   public final Counter slippedPutAttemptCount;
+  public final Counter metadataSlippedPutAttemptCount;
```
Comment (JingQianCloud) [inline ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouterMetrics.java:187]: Can we add comment for 
    slippedPutAttempCount,
    slippedPutSuccessCount,
    metadataSlippedPutSuccessCount
    metadataPutAttemptCount.

Looks like your intention is that slippedPutXXX includes both data chunk and meta data while metadataXXX is as name indicated.

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java [hunk 2]
```diff
@@ -580,14 +580,10 @@ private void onChunkOperationComplete(PutChunk chunk) {
       logger.trace("{}: Successfully put data chunk with blob id : {}", loggingContext, chunk.getChunkBlobId());
```
Comment (JingQianCloud) [inline ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:580]: I believe you don't need this log any more.

File: ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouterMetrics.java (hunk 1 above)
Comment (ankagrawal) [reply ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouterMetrics.java:189]: You are right about the intention.
Added the comments.

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java (hunk 2 above)
Comment (ankagrawal) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java]: fixed.

**Review by JingQianCloud — APPROVED** [review]

**Pull request merged**